### ISSUE TITLE
Executing batch script via command line hangs

### ISSUE THREAD

**Issue by mu88** [microsoft/terminal#3736]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.18362.476]
Windows Terminal version: 0.7.3291.0
```

# Steps to reproduce

1. Configure a profile with a batch file like this:
```
"name" : "Build MyApp",
"guid" : "{65869F08-7651-44A5-9EAB-982A1577BC65}",
"commandline" : "Batch\\Build.MyApp.bat",
"startingDirectory" : "C:\\Projects\\MyApp",
"closeOnExit" : false,
"fontFace" : "Fira Code",
"fontSize" : 10,
"icon" : "C:\\Projects\\MyApp\\MyApp.ico"
```
2. Execute the profile `Build MyApp`

# Expected behavior

The batch file `C:\Projects\MyApp\Batch\Build.MyApp.bat` will be executed.

# Actual behavior

A new tab will be opened, but nothing gets executed. It is hanging:
![image](https://user-images.githubusercontent.com/4560672/69708959-d2a43000-10fc-11ea-8bd2-c76ea570da78.png)

**Comment by zadjii-msft** [comment]
@mu88 what if you set the `commandline` to `cmd.exe /k Batch\\Build.MyApp.bat`? I think you need the `cmd.exe /k` to tell it to actually run cmd.exe with the batch script you've specified.

**Comment by mu88** [comment]
@zadjii-msft That actually solved my issue. Thank you very much! Do you have any idea why that popped up after upgrading Windows Terminal to the latest version?

**Comment by zadjii-msft** [comment]
Honestly I don't think it should have ever worked in the first place tbh. If I had to guess, it's because we changed from using `conhost` directly to using the pseudoconsole API properly. With the conhost backend, conhost would try and create the process on our behalf, but I guess it was able to `CreateProcess` a bat file? Then when we switched to conpty, we started creating the child process ourself, and maybe we weren't as liberal with what we could create.

/cc @DHowett-MSFT for visibility, because he likes these kinds of issues

**Comment by DHowett-MSFT** [comment]
The real issue here is that of the _working directory_. :smile:

The profile specifies a commandline of `Batch\Build.MyApp.bat` with a starting directory. When we were launching through conhost, we would launch conhost in the starting directory. When conhost went to launch `Build.MyApp.bat`, CreateProcess started searching relative to conhost's CWD.

Now that we're handing starting directory directly to the process we're spawning, instead of having an intermediate process change to that folder first, all commandlines will need to be either fully-qualified or supported by `%PATH%`.

TL;DR: Set the commandline to the full path to the batch file, or keep asking cmd to intervene (since we're starting cmd in the right folder, then asking cmd to find the batch file.)
